### PR TITLE
Complete transaction to frontend integration

### PR DESCRIPTION
**Pull request by trevorbernard**
Fixes: https://github.com/Granola-Team/mina-indexer/issues/848
Fixes: https://github.com/Granola-Team/mina-indexer/issues/852

* Add efficient filter by block height
* Use nanomina for transaction amount and fee

### CODE REVIEW

File: rust/src/web/graphql/transactions/mod.rs [hunk 1]
```diff
@@ -59,6 +60,20 @@ impl TransactionsQueryRoot {
             }
         };
 
+        // block height query
+        if let Some(block_height) = query.block_height {
+            let mut transactions: Vec<TransactionWithBlock> = db
+                .get_blocks_at_height(block_height as u32)?
```
Comment (jhult) [inline rust/src/web/graphql/transactions/mod.rs:66]: Should we (somehow) support providing a limit when querying the database?

I suppose we can't do that due to sorting.

File: tests/hurl/transactions.hurl [hunk 2]
```diff
@@ -40,3 +40,49 @@ jsonpath "$.data.transactions[49].memo" == "E4YM2vTHhWEg66xpj52JErHUBU4pZ1yageL4
...
+  "query": {
+    "canonical": true,
+    "blockHeight": 11
+  }
+}
+```
+
+HTTP 200
+[Asserts]
+jsonpath "$.data.transactions" count == 2
```
Comment (jhult) [inline tests/hurl/transactions.hurl:81]:
```suggestion
duration < 1000
jsonpath "$.data.transactions" count == 2
```

**Review by jhult — APPROVED** [review]
LGTM. Nicely done!

File: rust/src/web/graphql/transactions/mod.rs (hunk 1 above)
Comment (trevorbernard) [reply rust/src/web/graphql/transactions/mod.rs:66]: This won't ever be an expensive call. I've only ever seen 2-3 blocks per block height and there are a max of 128 txns per block.

**Review by jhult — APPROVED** [review]

**Pull request merged**